include/forcing/NetCDFPerFeatureDataProvider.hpp

```diff
@@ -0,0 +1,457 @@
+         * @throws std::out_of_range If the given point is not in any time step.
+         */
+        size_t get_ts_index_for_time(const time_t &epoch_time)
+        {
+            if (start_time <= epoch_time && epoch_time < stop_time)
+            {
+                double offset = epoch_time - start_time;
+                offset /= time_stride;
+                return size_t(offset);
+            }
+            else
+            {
+                std::stringstream ss;
+                ss << "The value " << epoch_time << "was not in the range [" << start_time << "," << stop_time << ")\n";
+                throw std::out_of_range(ss.str().c_str());
+            }
+        }
+
+        /**
+         * Get the value of a forcing property for an arbitrary time period, converting units if needed.
+         *
+         * An @ref std::out_of_range exception should be thrown if the data for the time period is not available.
+         *
+         * @param selector Data required to establish what subset of the stored data should be accessed
+         * @param variable_name The name of the data property of interest.
+         * @param init_time The epoch time (in seconds) of the start of the time period.
+         * @param duration_seconds The length of the time period, in seconds.
+         * @param output_units The expected units of the desired output value.
+         * @param m How data is to be resampled if there is a mismatch in data alignment or repeat rate
+         * @return The value of the forcing property for the described time period, with units converted if needed.
+         * @throws std::out_of_range If data for the time period is not available.
+         */
+        double get_value(Catchment_Id selector, const std::string &variable_name, const time_t &init_time, const long &duration_s,
+                                 const std::string &output_units, ReSampleMethod m)
+        {
+            auto stop_time = init_time + duration_s;
+            
+            auto idx1 = get_ts_index_for_time(init_time);
+            auto idx2 = get_ts_index_for_time(stop_time);
+
+            auto stride = idx2 - idx1;
+
+            std::vector<std::size_t> start, count;
+
+            auto cat_pos = id_pos[selector];
+
+            start.push_back(cat_pos);
+            start.push_back(idx1);
+
+            count.push_back(1);
+
+            double t1 = time_vals[idx1];
+            double t2 = time_vals[idx2];
+
+            double rvalue = 0.0;
+
+            auto ncvar = nc_file->getVar(variable_name);
+
+            std::string native_units;
+            
+            auto units_att = ncvar.getAtt("units");
+            if ( units_att.isNull() )
+            {
+                native_units = "unknown";      
+            }
+            else
+            {
+                units_att.getValues(native_units);
+            }
+
+            // case where less than one time step is requested
```

Comment:
If there is a general solution, drop these cases.  From a cursory look, the general algorithm should work fine in both these cases.

Reply:
Will update